railgun/internal/ctrlutils/ingress-status.go

```diff
@@ -0,0 +1,253 @@
+	routeInf := strings.Split(*(svc.Routes[0].Name), ".")
+	namespace = routeInf[0]
+	name = routeInf[1]
+	if len(namespace) == 0 || len(name) == 0 {
+		return fmt.Errorf("configured route information is not completed which should not happen.")
+	}
+
+	knativeCli, err := knativeversioned.NewForConfig(kubeCfg)
+	if err != nil {
+		return fmt.Errorf("failed to generate knative client. err %v", err)
+	}
+	ingClient := knativeCli.NetworkingV1alpha1().Ingresses(namespace)
+	curIng, err := ingClient.Get(ctx, name, metav1.GetOptions{})
+	if err != nil || curIng == nil {
+		return fmt.Errorf("failed to fetch Knative Ingress %v/%v: %w", namespace, name, err)
+	}
+
+	// check if CR current status already updated
+	var status []apiv1.LoadBalancerIngress
+	sort.SliceStable(status, lessLoadBalancerIngress(status))
+	curIPs := toCoreLBStatus(curIng.Status.PublicLoadBalancer)
+	ips, err := RunningAddresses(ctx, kubeCfg)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve cluster loadbalancer.")
+	}
+	status = SliceToStatus(ips)
+	if IngressSliceEqual(status, curIPs) &&
+		curIng.Status.ObservedGeneration == curIng.GetObjectMeta().GetGeneration() {
+		log.Info("no change in status, update skipped")
+		return nil
+	}
+
+	// updating current custom status
+	log.Info("attempting to update Knative Ingress status")
+	lbStatus := toKnativeLBStatus(status)
+	clusterDomain := network.GetClusterDomainName()
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(lbStatus); i++ {
+		lbStatus[i].DomainInternal = fmt.Sprintf("%s.%s.svc.%s",
+			"ingress-controller-kong-proxy", "kong-system", clusterDomain)
+	}
+
+	curIng.Status.MarkLoadBalancerReady(lbStatus, lbStatus)
+	ingressCondSet.Manage(&curIng.Status).MarkTrue(knative.IngressConditionReady)
+	ingressCondSet.Manage(&curIng.Status).MarkTrue(knative.IngressConditionNetworkConfigured)
+	curIng.Status.ObservedGeneration = curIng.GetObjectMeta().GetGeneration()
+
+	_, err = ingClient.UpdateStatus(ctx, curIng, metav1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to update ingress status: %v", err)
+	} else {
+		logger.Info("successfully updated ingress status")
+	}
+	return nil
+}
+
+func RunningAddresses(ctx context.Context, kubeCfg *rest.Config) ([]string, error) {
+	addrs := []string{}
+
+	namespace := "kong-system"
+	CoreClient, _ := clientset.NewForConfig(kubeCfg)
+	svc, err := CoreClient.CoreV1().Services(namespace).Get(ctx, "ingress-controller-kong-proxy", metav1.GetOptions{})
+	if err != nil {
+		log.Infof("running address err %v", err)
+		return nil, err
+	}
+
+	switch svc.Spec.Type {
+	case apiv1.ServiceTypeLoadBalancer:
+		for _, ip := range svc.Status.LoadBalancer.Ingress {
+			if ip.IP == "" {
+				addrs = append(addrs, ip.Hostname)
+			} else {
+				addrs = append(addrs, ip.IP)
+			}
+		}
+
+		addrs = append(addrs, svc.Spec.ExternalIPs...)
+		return addrs, nil
+	default:
+		return addrs, nil
+	}
+}
+
+// sliceToStatus converts a slice of IP and/or hostnames to LoadBalancerIngress
```

Comment:
```suggestion
// SliceToStatus converts a slice of IP and/or hostnames to LoadBalancerIngress
```

Reply:
Ditto godoc compliance. If these get closed without changes (or there's some other change elsewhere that invalidates them), they should have comments indicating why.